spec/matcher_spec.rb

```diff
@@ -168,6 +190,28 @@ def match_metadata(*rule)
       end
     end
 
+    describe "str" do
+      it "should behave for valid integers" do
+        match("=", ["str", 9      ], "9" ).should == true
+        match("=", ["str", "10"   ], "0" ).should == false
+        match("=", ["str", "0xf"  ], "0xf").should == true
+        match("=", ["str", "0b110"], "0b110" ).should == true
+        match("=", ["str", "027"  ], "027").should == true
+      end
+
+      it "should behave for valid floats" do
+        match("=", ["str", 5.4  ], "5.4").should == true
+        match("=", ["str", "2.7"], "2.7").should == true
+        match("=", ["str", "1e5"], "1e5").should == true
+      end
+
+      it "should behave for valid booleans and nil" do
+        match("=", ["str", true ], "true").should == true
+        match("=", ["str", false], "false").should == true
+        match("=", ["str", nil], '').should == true
+      end
+    end
```

Comment:
Not at this time. We have no complex objects or collections that can be used as arguments.

Perhaps worth noting, structured facts/metadata will throw a wrench in the matcher, but that's an [existing issue](https://tickets.puppetlabs.com/browse/RAZOR-641).